libs/reactivex/Observable_MinOperator.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections;
+
+namespace Cusco.ReactiveX;
+
+public abstract partial class Observable<T>
+{
+  public Observable<T> Min(IComparer<T> comparer)
+  {
+    if (null == comparer) throw new ArgumentNullException(nameof(comparer));
+
+    Option<T> minValue = Option<T>.None();
+
+    return Observable.Create<T>(dispatchQueue, observer =>
+    {
+      return Subscribe(
+        onNext: value =>
+        {
+          minValue = minValue.Map(min => comparer.Compare(value, min) < 0 ? value : min).Or(value);
+        },
+        onError: observer.OnError,
+        onComplete: () =>
+        {
+          if (minValue.isSome)
+            observer.OnNext(minValue.Unwrap());
+          observer.OnCompleted();
+        });
+    });
+  }
+}
+
+public static class ObservableExtensions
+{
+  public static Observable<T> Min<T>(this Observable<T> observable) where T: IComparable
+  {
+    Option<T> minValue = Option<T>.None();
```

Comment:
State should not be shared with all subscribers, move this variable inside the Observable.Create lambda